.github/workflows/autobuild.yml

```diff
@@ -251,77 +233,81 @@ jobs:
+        if:                         >-
+                                    steps.step_cmd3_postbuild.outputs.artifact_1 != '' &&
+                                    steps.step_macos_build.outputs.macos_signed == 'true' &&
+                                    contains(needs.create_release.outputs.publish_to_release, 'true')
+        id:                         notarize-macOS-app
+        uses:                       devbotsxyz/xcode-notarize@d7219e1c390b47db8bab0f6b4fc1e3b7943e4b3b
         with:
-          product-path: deploy/${{ steps.step_cmd3_postbuild.outputs.artifact_1 }}
-          primary-bundle-id: io.jamulus.Jamulus
+          product-path:             deploy/${{ steps.step_cmd3_postbuild.outputs.artifact_1 }}
+          primary-bundle-id:        io.jamulus.Jamulus
           appstore-connect-username: ${{ secrets.NOTARIZATION_USERNAME }}
           appstore-connect-password: ${{ secrets.NOTARIZATION_PASSWORD }}
-      # Apply successful notarization
-      - name: "Staple macOS Release Build"
-        if: ${{ (steps.step_cmd3_postbuild.outputs.artifact_1 != '') && (steps.step_macos_build.outputs.macos_signed == 'true') && contains(needs.create_release.outputs.publish_to_release, 'true') }}
-        id: staple-macOS-app
-        uses: devbotsxyz/xcode-staple@v1
+
+      - name:                       "Staple macOS Release Build"
+        if:                         >-
+                                    steps.step_cmd3_postbuild.outputs.artifact_1 != '' &&
+                                    steps.step_macos_build.outputs.macos_signed == 'true' &&
+                                    contains(needs.create_release.outputs.publish_to_release, 'true')
+        id:                         staple-macOS-app
+        uses:                       devbotsxyz/xcode-staple@v1
         with:
-          product-path: deploy/${{ steps.step_cmd3_postbuild.outputs.artifact_1 }}
+          product-path:             deploy/${{ steps.step_cmd3_postbuild.outputs.artifact_1 }}
 
-      # Upload Artifact to Release
       - name:                       Upload Artifact 1 to Release
-        if:                         ${{ (steps.step_cmd3_postbuild.outputs.artifact_1 != '') && contains(needs.create_release.outputs.publish_to_release, 'true') }}
+        if:                         >-
+                                    steps.step_cmd3_postbuild.outputs.artifact_1 != '' &&
+                                    contains(needs.create_release.outputs.publish_to_release, 'true')
         id:                         upload-release-asset1
         uses:                       actions/upload-release-asset@v1
         env:
           GITHUB_TOKEN:             ${{ secrets.GITHUB_TOKEN }}
         with:
-          upload_url:               ${{ needs.create_release.outputs.upload_url }} # See this blog post for more info: https://jasonet.co/posts/new-features-of-github-actions/#passing-data-to-future-steps
```

Comment:
Maybe leave this link here and delete it below.

Reply:
Hm, I'd rather like to remove it. Reasoning:
- While this might have been a new feature at some point, it no longer is. It's standard Github Workflow syntax and not some kind of hard to find trick or something. Anyone who touches the workflow should read the documentation. I'm considering this to be a rather foundational thing and we don't link foundational things in other places either (e.g. C++ or Qt books in source files ;)).
- I don't know why it's worth to document this feature (the "outputs" logic) right here. The whole workflow is sprinkled with `outputs` references as it's such a basic idiom. So if we document it at all, it should be documented at the very top of the file. Maybe I'm missing something, but `grep -cF '.outputs.' .github/workflows/autobuild.yml` returns 33 places where outputs are either set or read. I don't see why those two occurences would be special.

Reply:
> Of course the "can also be warn,…" can be removed but the links to the documentation or blog posts should remain.

(I've replied to that directly in the other thread)

> Some comments might be trivial (Checkout code,…)

I'm a huge fan of comments to explain specialities. I'm annoyed by comments which state the obvious. They just make the file longer and make me think "umm, I've just read that already". It's a [code smell](https://moderatemisbehaviour.github.io/clean-code-smells-and-heuristics/comments/c3-redundant-comment.html).
You've already mentioned one extreme case. Here's another.
https://github.com/jamulussoftware/jamulus/blob/master/.github/workflows/autobuild.yml#L238-L239

Thanks for the specific comments, that's much easier to work with. :)

Reply:
Ok. Probably I still think of the old times where the .yaml was basically only a three step procedure (and where I wanted it to be clarified by nefarius). 

So I'll need to be happy with it.

Reply:
> So I'll need to be happy with it.

Well, I don't want to say "it must be like that". I want to outline what my goals are. If I'm missing something or if there are better approaches, I'm open to that. But it should have some justification :)

Reply:
The thing is  - in my opinion - the whole autobuild file has become complicated (there are many specialities. Let me do another review on how I would comment the code)